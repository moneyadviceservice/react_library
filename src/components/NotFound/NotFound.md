Renders information and useful links when pages are not found.

```jsx
import React, { useState } from 'react';
import { Row, Button, NotFound } from '@moneypensionservice/directories';

const [locale, setLocale] = useState('en');

<>
<Row margin={{bottom: '20px'}}>
  <Button onClick={() => setLocale('en')}>English</Button>
  <Button onClick={() => setLocale('cy')}>Welsh</Button>
</Row>
<Row justify="center">
  <NotFound
    currentLng={locale} />
</Row>
</>
```
