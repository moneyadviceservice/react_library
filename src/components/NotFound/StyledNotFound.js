import styled from 'styled-components'
import { Row, Col } from '../Grid'
import { Heading } from '../Heading'

const NotFoundContainer = styled(Row)`
  overflow: hidden;
  min-height: 650px;
`

const NotFoundSection = styled(Col)``

const NotFoundHeading = styled(Heading)`
  color: ${({ theme }) => theme.colors.notFound.heading};
`
const LinksHeading = styled(Heading)`
  font-weight: 700;
  margin: 0 0 1.5em;
  font-size: 1.25em;
  color: ${({ theme }) => theme.colors.notFound.linksHeading};
`

export { NotFoundContainer, NotFoundSection, NotFoundHeading, LinksHeading }
