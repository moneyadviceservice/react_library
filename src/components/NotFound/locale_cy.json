{
  "heading": "Mae'n ddrwg gennym. Ni allwn ddod o hyd i'r dudalen honno.",
  "subHeading": "Gwiriwch y cyfeiriad y gwnaethoch ei nodi neu ceisiwch eto yn nes ymlaen a dylech allu dod o hyd iddo.",
  "links": {
    "title": "Dolenni defnyddiol",
    "link_1": {
      "text": "Dychwelwch i hafan y Cyfeiriadur Cyngor Teithio",
      "url": "https://traveldirectory.moneyadviceservice.org.uk/cy"
    },
    "link_2": {
      "text": "Ewch i hafan y Gwasanaeth Cyngor Arian",
      "url": "https://www.moneyadviceservice.org.uk/cy"
    }
  }
}
