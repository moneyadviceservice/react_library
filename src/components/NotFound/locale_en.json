{
  "heading": "We're sorry. We can't find that page.",
  "subHeading": "Please check the address you entered or try again later and you should be able to find it.",
  "links": {
    "title": "Useful links",
    "link_1": {
      "text": "Return to the Travel Advice Directory homepage",
      "url": "https://traveldirectory.moneyadviceservice.org.uk/en"
    },
    "link_2": {
      "text": "Go to the Money Advice Service homepage",
      "url": "https://www.moneyadviceservice.org.uk/en"
    }
  }
}
