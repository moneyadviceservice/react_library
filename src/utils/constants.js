export const breakpoints = {
  xs: 1,
  sm: 48, // 768px
  md: 64, // 1024px
  lg: 90, // 1440px
  xl: 120, // 1920px
}

export const dimensions = ['xs', 'sm', 'md', 'lg', 'xl']

export const mediaQuery = 'only screen'

export const ALIGN_SELF_MAP = {
  center: 'center',
  end: 'flex-end',
  start: 'flex-start',
  stretch: 'stretch',
}

export const SPACING_SIZES = [
  'xxsmall',
  'xsmall',
  'small',
  'medium',
  'large',
  'xlarge',
]

