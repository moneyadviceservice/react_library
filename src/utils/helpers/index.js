export * from './background'
export * from './responsive'
export * from './flexbox'
export * from './styles'
export * from './mediaQueries'
export * from './sizing'
export * from './refs'
