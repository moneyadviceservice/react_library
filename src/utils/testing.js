import { render, act } from '@testing-library/react'
import GlobalTheme from '../components/ThemeProvider'

const customRender = (ui, options) =>
  render(ui, { wrapper: GlobalTheme, ...options })

const waitRender = async (ui, options) => {
  const { container, ...rest } = customRender(ui, options)

  await act(async () => {
    await Promise.resolve(container)
  })

  return { container, ...rest }
}

export * from '@testing-library/react'

export { customRender as render, waitRender }
