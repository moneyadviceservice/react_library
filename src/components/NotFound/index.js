import React from 'react'
import PropTypes from 'prop-types'
import { genericPropTypes, genericPropsDefaults } from '../../utils/prop-types'
// components
import {
  NotFoundContainer,
  NotFoundSection,
  NotFoundHeading,
  LinksHeading,
} from './StyledNotFound'
import { Paragraph as P } from '../Paragraph'
import { Anchor } from '../Anchor'
// translations
import LocaleEn from './locale_en.json'
import LocaleCy from './locale_cy.json'

const NotFound = ({
  a11yTitle,
  as,
  children,
  currentLng,
  linkText,
  LinkUrl,
  i18nLng,
  ...rest
}) => {
  // locale
  const lng = i18nLng || (currentLng === 'cy' ? LocaleCy : LocaleEn)

  return (
    <NotFoundContainer
      aria-label={a11yTitle}
      constrained
      direction="column"
      forwardedAs={as}
      justify="space-evenly"
      {...rest}
    >
      <NotFoundSection grow={false}>
        <NotFoundHeading level={1}>{lng.heading}</NotFoundHeading>
        <P>{lng.subHeading}</P>
      </NotFoundSection>
      <NotFoundSection grow={false}>
        <LinksHeading level={2}>{lng.links.title}</LinksHeading>
        <Anchor data-testid="url" href={LinkUrl || lng.links.link_1.url}>
          {linkText || lng.links.link_1.text}
        </Anchor>
        <Anchor href={lng.links.link_2.url}>{lng.links.link_2.text}</Anchor>
      </NotFoundSection>
      {children && <NotFoundSection grow={false}>{children}</NotFoundSection>}
    </NotFoundContainer>
  )
}

// Documentation
NotFound.propTypes = {
  /** The DOM tag or react component to use for the element. */
  as: PropTypes.oneOfType([PropTypes.string, PropTypes.node, PropTypes.func]),
  /** Current Language Value */
  currentLng: PropTypes.oneOf(['en', 'cy']),
  /** Alternate text to be displayed in the directory link. */
  linkText: PropTypes.string,
  /** Alternate directory link url. */
  LinkUrl: PropTypes.string,
  /** Alternate translations for the card. */
  i18nLng: PropTypes.object,
  ...genericPropTypes,
}

NotFound.defaultProps = {
  as: 'section',
  currentLng: 'en',
  ...genericPropsDefaults(),
}

/** @component */
export { NotFound }
