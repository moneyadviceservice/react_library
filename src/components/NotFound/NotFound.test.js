import React from 'react'
import { render } from '../../utils/testing'
import { NotFound } from '../NotFound'

describe('<NotFound />', () => {
  describe('Snapshots', () => {
    it('should match snapshot', () => {
      const { container } = render(<NotFound />)
      expect(container).toMatchSnapshot()
    })
    it('should match snapshot in Welsh', () => {
      const { container } = render(<NotFound currentLng="cy" />)
      expect(container).toMatchSnapshot()
    })
  })

  describe('custom link props', () => {
    it('linkText and linkUrl props', () => {
      const { container, getByTestId } = render(
        <NotFound linkText="customLinkText" linkUrl="https://www.test.com/" />
      )
      // snapshot
      expect(container).toMatchSnapshot()
      // anchor text content
      expect(getByTestId('url').textContent).toBe('customLinkText')
      // href
      expect(getByTestId('url').href).toBe('https://www.test.com/')
    })
  })
})
