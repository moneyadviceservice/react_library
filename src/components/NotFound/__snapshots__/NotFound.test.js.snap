// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> Snapshots should match snapshot 1`] = `
<div>
  <section
    class="styledRow__RowWrapper-sc-1ip85in-0 kGwesl StyledNotFound__NotFoundContainer-v06e70-0 ghBYdL"
  >
    <div
      class="styledCol__ColWrapper-sc-1mivnjm-0 kcBKMp StyledNotFound__NotFoundSection-v06e70-1"
    >
      <h1
        class="StyledHeading-xxi4nn-0 likqsL StyledNotFound__NotFoundHeading-v06e70-2 cNoFKm"
      >
        We're sorry. We can't find that page.
      </h1>
      <p
        class="StyledParagraph-sc-1toaej-0 gpltyO"
      >
        Please check the address you entered or try again later and you should be able to find it.
      </p>
    </div>
    <div
      class="styledCol__ColWrapper-sc-1mivnjm-0 kcBKMp StyledNotFound__NotFoundSection-v06e70-1"
    >
      <h2
        class="StyledHeading-xxi4nn-0 boqBEm StyledNotFound__LinksHeading-v06e70-3 eClJdW"
      >
        Useful links
      </h2>
      <a
        class="StyledParagraph-sc-1toaej-0 gpltyO StyledAnchor-sc-1qj1ci5-0 fNhjgf"
        data-testid="url"
        href="https://traveldirectory.moneyadviceservice.org.uk/en"
      >
        Return to the Travel Advice Directory homepage
      </a>
      <a
        class="StyledParagraph-sc-1toaej-0 gpltyO StyledAnchor-sc-1qj1ci5-0 fNhjgf"
        href="https://www.moneyadviceservice.org.uk/en"
      >
        Go to the Money Advice Service homepage
      </a>
    </div>
  </section>
</div>
`;

exports[`<NotFound /> Snapshots should match snapshot in Welsh 1`] = `
<div>
  <section
    class="styledRow__RowWrapper-sc-1ip85in-0 kGwesl StyledNotFound__NotFoundContainer-v06e70-0 ghBYdL"
  >
    <div
      class="styledCol__ColWrapper-sc-1mivnjm-0 kcBKMp StyledNotFound__NotFoundSection-v06e70-1"
    >
      <h1
        class="StyledHeading-xxi4nn-0 likqsL StyledNotFound__NotFoundHeading-v06e70-2 cNoFKm"
      >
        Mae'n ddrwg gennym. Ni allwn ddod o hyd i'r dudalen honno.
      </h1>
      <p
        class="StyledParagraph-sc-1toaej-0 gpltyO"
      >
        Gwiriwch y cyfeiriad y gwnaethoch ei nodi neu ceisiwch eto yn nes ymlaen a dylech allu dod o hyd iddo.
      </p>
    </div>
    <div
      class="styledCol__ColWrapper-sc-1mivnjm-0 kcBKMp StyledNotFound__NotFoundSection-v06e70-1"
    >
      <h2
        class="StyledHeading-xxi4nn-0 boqBEm StyledNotFound__LinksHeading-v06e70-3 eClJdW"
      >
        Dolenni defnyddiol
      </h2>
      <a
        class="StyledParagraph-sc-1toaej-0 gpltyO StyledAnchor-sc-1qj1ci5-0 fNhjgf"
        data-testid="url"
        href="https://traveldirectory.moneyadviceservice.org.uk/cy"
      >
        Dychwelwch i hafan y Cyfeiriadur Cyngor Teithio
      </a>
      <a
        class="StyledParagraph-sc-1toaej-0 gpltyO StyledAnchor-sc-1qj1ci5-0 fNhjgf"
        href="https://www.moneyadviceservice.org.uk/cy"
      >
        Ewch i hafan y Gwasanaeth Cyngor Arian
      </a>
    </div>
  </section>
</div>
`;

exports[`<NotFound /> custom link props linkText and linkUrl props 1`] = `
<div>
  <section
    class="styledRow__RowWrapper-sc-1ip85in-0 kGwesl StyledNotFound__NotFoundContainer-v06e70-0 ghBYdL"
  >
    <div
      class="styledCol__ColWrapper-sc-1mivnjm-0 kcBKMp StyledNotFound__NotFoundSection-v06e70-1"
    >
      <h1
        class="StyledHeading-xxi4nn-0 likqsL StyledNotFound__NotFoundHeading-v06e70-2 cNoFKm"
      >
        We're sorry. We can't find that page.
      </h1>
      <p
        class="StyledParagraph-sc-1toaej-0 gpltyO"
      >
        Please check the address you entered or try again later and you should be able to find it.
      </p>
    </div>
    <div
      class="styledCol__ColWrapper-sc-1mivnjm-0 kcBKMp StyledNotFound__NotFoundSection-v06e70-1"
    >
      <h2
        class="StyledHeading-xxi4nn-0 boqBEm StyledNotFound__LinksHeading-v06e70-3 eClJdW"
      >
        Useful links
      </h2>
      <a
        class="StyledParagraph-sc-1toaej-0 gpltyO StyledAnchor-sc-1qj1ci5-0 fNhjgf"
        data-testid="url"
        href="https://www.test.com/"
      >
        customLinkText
      </a>
      <a
        class="StyledParagraph-sc-1toaej-0 gpltyO StyledAnchor-sc-1qj1ci5-0 fNhjgf"
        href="https://www.moneyadviceservice.org.uk/en"
      >
        Go to the Money Advice Service homepage
      </a>
    </div>
  </section>
</div>
`;
